config/management/genesis/src/verify.rs

```diff
@@ -147,45 +147,35 @@ fn compare_genesis(
     storage: &Storage,
     buffer: &mut String,
     genesis_path: &PathBuf,
-    storage_name: &'static str,
 ) -> Result<(), Error> {
     // Compute genesis and waypoint and compare to given waypoint
     let db_path = TempPath::new();
     let (db_rw, expected_waypoint) = compute_genesis(genesis_path, db_path.path())?;
 
-    let actual_waypoint = storage
-        .get(WAYPOINT)
-        .and_then(|c| c.value.string())
-        .map_err(|e| Error::StorageReadError(storage_name, WAYPOINT, e.to_string()))?;
-    let actual_waypoint = Waypoint::from_str(&actual_waypoint).map_err(|e| {
-        Error::UnexpectedError(format!("Unable to parse waypoint: {}", e.to_string()))
-    })?;
+    let actual_waypoint = storage.waypoint(WAYPOINT)?;
     write_assert(buffer, WAYPOINT, actual_waypoint == expected_waypoint);
 
     // Fetch on-chain validator config and compare on-chain keys to local keys
-    let validator_account = validator_account(storage, storage_name)?;
+    let validator_account = storage.account_address(OWNER_ACCOUNT)?;
     let validator_config = validator_config(validator_account, db_rw.reader.clone())?;
 
-    let actual_consensus_key = ed25519_from_storage(CONSENSUS_KEY, storage)
-        .map_err(|e| Error::StorageReadError(storage_name, CONSENSUS_KEY, e))?;
+    let actual_consensus_key = storage.ed25519_public_from_private(CONSENSUS_KEY)?;
     let expected_consensus_key = validator_config.consensus_public_key;
     write_assert(
         buffer,
         CONSENSUS_KEY,
         actual_consensus_key == expected_consensus_key,
     );
 
-    let actual_validator_key = x25519_from_storage(VALIDATOR_NETWORK_KEY, storage)
-        .map_err(|e| Error::StorageReadError(storage_name, VALIDATOR_NETWORK_KEY, e))?;
+    let actual_validator_key = storage.x25519_public_from_private(VALIDATOR_NETWORK_KEY)?;
     let expected_validator_key = validator_config.validator_network_identity_public_key;
     write_assert(
         buffer,
         VALIDATOR_NETWORK_KEY,
         actual_validator_key == expected_validator_key,
     );
 
-    let actual_fullnode_key = x25519_from_storage(FULLNODE_NETWORK_KEY, storage)
-        .map_err(|e| Error::StorageReadError(storage_name, FULLNODE_NETWORK_KEY, e))?;
+    let actual_fullnode_key = storage.x25519_public_from_private(FULLNODE_NETWORK_KEY)?;
```

Comment:
Yay! less error handling, this was in my list of refactors I was going to get back to :)